plugins/net_plugin/net_plugin.cpp

```diff
@@ -3484,38 +3473,40 @@ namespace eosio {
    // called from any thread
    void net_plugin_impl::update_security_group(const block_state_ptr& bs) {
       // update cache
+      //    - connection_mtx is needed
       //
       auto& update = bs->get_security_group_info();
+      std::shared_lock<std::shared_mutex> connection_guard(connections_mtx);
       if(!security_group.update_cache(update.version, update.participants)) {
          return;
       }
 
       std::vector<connection_ptr> added_connections;
       added_connections.reserve(connections.size());
 
-      // update connection
+      // update connections
+      //    - connection_mtx still needed
       //
-      auto do_update = [&](auto& connection) {
+      for(auto& connection : connections) {
          const auto& participant = connection->participant_name();
          if(!participant) {
-            return true;
+            return;
```

Comment:
this should be continue instead of return

Reply:
changed to continue